ui/addon/src/main/java/org/jboss/windup/ui/RulesetUpdateChecker.java

```diff
@@ -0,0 +1,111 @@
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.forge.addon.dependencies.builder.DependencyQueryBuilder;
+import org.jboss.forge.furnace.Furnace;
+import org.jboss.forge.furnace.event.PostStartup;
+import org.jboss.forge.furnace.services.Imported;
+import org.jboss.windup.config.furnace.FurnaceHolder;
+import org.jboss.windup.util.PathUtil;
+
+@Singleton
+public class RulesetUpdateChecker
+{
+
+    private static final String META_INF_VERSION_KEY = "version=";
+    public static final String RULESET_CORE_DIRECTORY = "migration-core";
+
+    public void perform(@Observes PostStartup event)
+    {
+        boolean isDependencies = event.getAddon().getId().toString().contains("org.jboss.windup.ui:windup-ui");
+        if(isDependencies) {
+            Furnace furnace = FurnaceHolder.getFurnace();
+            Imported<DependencyResolver> exportedTypes = furnace.getAddonRegistry().getServices(DependencyResolver.class);
+            if(!exportedTypes.isUnsatisfied() && rulesetNeedUpdate(exportedTypes.get())) {
+                    System.out.println("");
+                    System.out.println("Your ruleset is obsolete. Consider running windup-update-ruleset or windup-update-distribution command. Press ENTER to continue.");
+                    System.out.println("");
+            }
+            
+        }
+    }
+
+    public static boolean rulesetNeedUpdate(DependencyResolver dependencyResolver)
+    {
+        List<Coordinate> resolveVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder.create(CoordinateBuilder.create()
+                    .setGroupId("org.jboss.windup.rules")
+                    .setArtifactId("windup-rulesets")));
+       
+        int i = 0;
+        Coordinate latestCoordinate;
+        do{
+            i++;
+            latestCoordinate = resolveVersions.get(resolveVersions.size() - i);
+        }while(latestCoordinate.isSnapshot()); 
+        Path windupRulesDir = PathUtil.getWindupRulesDir();
+        Path coreRulesPropertiesPath = windupRulesDir.resolve(RULESET_CORE_DIRECTORY
+                    + "/META-INF/maven/org.jboss.windup.rules/windup-rulesets/pom.properties");
+        try (BufferedReader br = Files.newBufferedReader(coreRulesPropertiesPath, StandardCharsets.UTF_8))
+        {
+            String line;
+
+            while ((line = br.readLine()) != null)
+            {
+                if (line.startsWith(META_INF_VERSION_KEY))
+                {
+                    String installedVersion = line.substring(META_INF_VERSION_KEY.length());
+                    String latestVersion = latestCoordinate.getVersion();
+                    return versionIsOld(installedVersion,latestVersion);
+                }
+            }
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+        return false;
+
+    }
+    
+    private static boolean versionIsOld(String installedVersion, String latestVersion) {
```

Comment:
Is there a reason you didn't use the org.jboss.forge.furnace.versions.Versions class here? I would update to use that instead if it is suitable for this use-case.